warning: #warning "FFMPEG does not support H264 encoding (GFX H264 server mode)

I compile for openSUSE Tumbleweed with:

-DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX:PATH=/usr -DCMAKE_INSTALL_LIBDIR:PATH=lib -DCMAKE_SKIP_INSTALL_RPATH=ON -DWITH_CUPS=ON -DWITH_CLIENT=ON -DWITH_PCSC=ON -DWITH_ALSA=ON -DWITH_PULSE=ON -DWITH_OSS=OFF -DWITH_DIRECTFB=OFF -DWITH_JPEG=ON -DWITH_GSM=ON -DWITH_GSTREAMER_1_0=ON -DWITH_IPP=OFF -DWITH_OPENSSL=ON -DWITH_CHANNELS=ON -DWITH_X11=ON -DWITH_XCURSOR=ON -DWITH_XEXT=ON -DWITH_XKBFILE=ON -DWITH_XI=ON -DWITH_XINERAMA=ON -DWITH_XRENDER=ON -DWITH_XV=ON -DWITH_ZLIB=ON -DWITH_KRB5=ON -DBUILTIN_CHANNELS=OFF -DWITH_SERVER=OFF -DWITH_SSE2=ON -DWITH_DEBUG_SCARD=OFF -DWITH_SMARTCARD_INSPECT=ON -DWITH_WAYLAND=ON -DWITH_LIBSYSTEMD=ON -DCMAKE_NO_BUILTIN_CHRPATH=ON 

Together with the available devel-libraries this results in:
-DWITH_FFMPEG=ON
-DWITH_X264=OFF
-DWITH_OPENH264=OFF 

Now I get this message:

[   55s] /home/abuild/rpmbuild/BUILD/FreeRDP-master/libfreerdp/codec/h264_ffmpeg.c:29:2: warning: #warning "FFMPEG does not support H264 encoding (GFX H264 server mode). Please review your configuration!" [-Wcpp]
[   55s]  #warning "FFMPEG does not support H264 encoding (GFX H264 server mode). Please review your configuration!"
[   55s]   ^~~~~~~

Is this expected with above options?
For legal reasons the openSUSE packages do not support any patent encumbered codecs.

--- a/channels/drive/client/drive_file.c
+++ b/channels/drive/client/drive_file.c
@@ -385,16 +385,13 @@ BOOL drive_file_free(DRIVE_FILE* file)
 
 BOOL drive_file_seek(DRIVE_FILE* file, UINT64 Offset)
 {
-	LONG lDistHigh;
-	DWORD dwPtrLow;
+	LARGE_INTEGER loffset;
 
 	if (!file)
 		return FALSE;
 
-	lDistHigh = Offset >> 32;
-	DEBUG_WSTR("Seek %s", file->fullpath);
-	dwPtrLow = SetFilePointer(file->file_handle, Offset & 0xFFFFFFFF, &lDistHigh, FILE_BEGIN);
-	return dwPtrLow != INVALID_SET_FILE_POINTER;
+	loffset.QuadPart = Offset;
+	return SetFilePointerEx(file->file_handle, loffset, NULL, FILE_BEGIN);
 }
 
 BOOL drive_file_read(DRIVE_FILE* file, BYTE* buffer, UINT32* Length)
@@ -623,8 +620,7 @@ BOOL drive_file_set_information(DRIVE_FILE* file, UINT32 FsInformationClass, UIN
 
 			liSize.QuadPart = size & 0xFFFFFFFF;
 
-			if (SetFilePointer(file->file_handle, liSize.LowPart, &liSize.HighPart,
-			                   FILE_BEGIN) == INVALID_SET_FILE_POINTER)
+			if (!SetFilePointerEx(file->file_handle, liSize, NULL, FILE_BEGIN))
 			{
 				WLog_ERR(TAG, "Unable to truncate %s to %d (%"PRId32")", file->fullpath, size, GetLastError());
 				return FALSE;

--- a/winpr/libwinpr/file/file.c
+++ b/winpr/libwinpr/file/file.c
@@ -122,11 +122,18 @@ static DWORD FileSetFilePointer(HANDLE hFile, LONG lDistanceToMove,
 			PLONG lpDistanceToMoveHigh, DWORD dwMoveMethod)
 {
 	WINPR_FILE* pFile = (WINPR_FILE*) hFile;
-	long offset = lDistanceToMove;
+	INT64 offset;
 	int whence;
 
 	if (!hFile)
 		return INVALID_SET_FILE_POINTER;
+
+	if (lpDistanceToMoveHigh)
+	{
+		offset = (*lpDistanceToMoveHigh << 32) | (UINT32)lDistanceToMove;
+	}
+	else
+		 offset = lDistanceToMove;
 
 	switch(dwMoveMethod)
 	{
@@ -151,6 +158,42 @@ static DWORD FileSetFilePointer(HANDLE hFile, LONG lDistanceToMove,
 	}
 
 	return ftell(pFile->fp);
+}
+
+static BOOL FileSetFilePointerEx(HANDLE hFile, LARGE_INTEGER liDistanceToMove, PLARGE_INTEGER lpNewFilePointer, DWORD dwMoveMethod)
+{
+	WINPR_FILE* pFile = (WINPR_FILE*) hFile;
+	int whence;
+
+	if (!hFile)
+		return FALSE;
+
+	switch(dwMoveMethod)
+	{
+	case FILE_BEGIN:
+		whence = SEEK_SET;
+		break;
+	case FILE_END:
+		whence = SEEK_END;
+		break;
+	case FILE_CURRENT:
+		whence = SEEK_CUR;
+		break;
+	default:
+		return FALSE;
+	}
+
+	if (fseek(pFile->fp, liDistanceToMove.QuadPart, whence))
+	{
+		WLog_ERR(TAG, "fseek(%s) failed with %s [0x%08X]", pFile->lpFileName,
+			 strerror(errno), errno);
+		return FALSE;
+	}
+
+	if (lpNewFilePointer)
+		lpNewFilePointer->QuadPart = ftell(pFile->fp);
+
+	return TRUE;
 }
 
 static BOOL FileRead(PVOID Object, LPVOID lpBuffer, DWORD nNumberOfBytesToRead,
@@ -550,7 +593,7 @@ static HANDLE_OPS fileOps = {
 	NULL, /*  FlushFileBuffers */
 	FileSetEndOfFile,
 	FileSetFilePointer,
-	NULL, /* SetFilePointerEx */
+	FileSetFilePointerEx,
 	NULL, /* FileLockFile */
 	FileLockFileEx,
 	FileUnlockFile,

